src/components/tide-countdown.tsx

```diff
@@ -0,0 +1,97 @@
+import { Card, CardContent } from "@/components/ui/card";
+import { useEffect, useState } from "react";
+
+import { Separator } from "@/components/ui/separator";
+import { WavesLadder } from "lucide-react";
+
+interface TideCountdownProps {
+  nextTideTime: Date | "tomorrow";
+  nextTideType: "low" | "high";
+}
+
+export function TideCountdown({
+  nextTideTime,
+  nextTideType,
+}: TideCountdownProps) {
+  const [timeLeft, setTimeLeft] = useState<{
+    hours: string;
+    minutes: string;
+    seconds: string;
+  }>({
+    hours: "00",
+    minutes: "00",
+    seconds: "00",
+  });
+
+  useEffect(() => {
```

Comment:
Vercel's v0 spit this code out. I think it's an okay starting point, but -- generally -- for animations and regular updates like this, doing it in a `useEffect` gives me heebilyjeebilies. I want to refactor this to use `requestAnimationFrame` + `useRef` for a more accurate countdown. Also reduces the "footgun-y-ness" of this code.